### ISSUE TITLE
aws_msk_cluster : default value for client_broker : "TLS"

### ISSUE THREAD

**Issue by kunsalvi** [terraform-providers/terraform-provider-aws#10673]
Hi 

I am creating MSK cluster and if i do not keep any  encryption_in_transit. It picks up "TLS" value for client_broker and cluster is created. But if I do terrform apply (without making any changes.). It forces below change : 
      encryption_info.0.encryption_in_transit.0.client_broker: "TLS" => "TLS_PLAINTEXT" (forces new resource)

Moreover if I provide value as : 
    encryption_in_transit {
    client_broker = "TLS"
    in_cluster  = true
  }

Cluster creation fails with : 
Error: Error refreshing state: 1 error occurred:
	* aws_msk_cluster.atropos_prod: 1 error occurred:
	* aws_msk_cluster.atropos_prod: aws_msk_cluster.XXXX: failed requesting bootstrap broker info for "arn:aws:kafka:ap-south-1:286817589435:cluster/XXXX/684bd0d9-eecf-454d-b209-306ead63dda6-4" : BadRequestException: You can't get bootstrap broker nodes for a cluster in FAILED state.
	status code: 400, request id: 81a984a9-2646-46e7-86af-236c0a88b329

Terrform version : Terraform v0.11.14

**Comment by kunsalvi** [comment]
@bflad 

Thank you for the response. For now, I am explicitly specifying "client_broker" and its working for me. 

And "BadRequestException" is due to my certificate issue.

**Comment by wvidana** [comment]
This is even worst now.

If not specified it forces a new resource since now it wants to be `null`

```terraform
          - encryption_in_transit {
              - client_broker = "TLS" -> null # forces replacement
              - in_cluster    = true -> null
            }
```

**Comment by bflad** [comment]
The attribute default update has been merged and will release with version 3.0.0 of the Terraform AWS Provider, likely in the next two weeks. 👍
